components/net/bluetooth_thread.rs

```diff
@@ -242,6 +249,38 @@ impl BluetoothManager {
         }
     }
 
+
+    fn is_id_still_in_context(&self, id: &str, id_level: GATTHierarchyLevel) -> bool {
+        loop {
+            match id_level {
+                GATTHierarchyLevel::Characteristic => {
+                    if self.cached_characteristics.contains_key(id) {
+                        if let Some(service_id) = self.characteristic_to_service.get(id) {
+                            return self.is_id_still_in_context(service_id, GATTHierarchyLevel::Service)
+                        }
+                        return false;
+                    }
+                    return false;
+                },
+                GATTHierarchyLevel::Service => {
+                    if self.cached_services.contains_key(id) {
+                        if let Some(device_id) = self.service_to_device.get(id) {
+                            return self.is_id_still_in_context(device_id, GATTHierarchyLevel::Device)
+                        }
+                        return false;
+                    }
+                    return false;
+                },
+                GATTHierarchyLevel::Device => {
+                    if self.cached_devices.contains_key(id) {
```

Comment:
I really like this `if something true then return true, otherwise return false` code snippet :)